statuspage_connector.py

```diff
@@ -0,0 +1,591 @@
+# File: statuspage_connector.py
+#
+# Copyright (c) 2022 Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific language governing permissions
+# and limitations under the License.
+#
+#
+import json
+import sys
+
+import phantom.app as phantom
+import requests
+from bs4 import BeautifulSoup
+from phantom.action_result import ActionResult
+from phantom.base_connector import BaseConnector
+
+from statuspage_consts import *
+
+
+class RetVal(tuple):
+    def __new__(cls, val1, val2):
+        return tuple.__new__(RetVal, (val1, val2))
+
+
+class StatuspageConnector(BaseConnector):
+
+    def __init__(self):
+
+        # Call the BaseConnectors init first
+        super(StatuspageConnector, self).__init__()
+
+        self._state = None
+
+        # Variable to hold a base_url in case the app makes REST calls
+        # Do note that the app json defines the asset config, so please
+        # modify this as you deem fit.
+        self._base_url = None
+
+        self._api_key = None
+
+    def _get_error_message_from_exception(self, e):
+        """
+        Get appropriate error message from the exception.
+
+        :param e: Exception object
+        :return: error message
+        """
+        error_code = None
+        error_msg = ERR_MSG_UNAVAILABLE
+
+        try:
+            if hasattr(e, "args"):
+                if len(e.args) > 1:
+                    error_code = e.args[0]
+                    error_msg = e.args[1]
+                elif len(e.args) == 1:
+                    error_msg = e.args[0]
+        except Exception:
+            pass
```

Comment:
Instead of "pass", we can add a debug statement, to understand the error from logs.
For example, self.debug_print("Error occurred while fetching exception information")